### PR TITLE
Louis branch 1

### PR DESCRIPTION
**Pull request by lacattano**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: todo_app/app.py [hunk 1]
```diff
@@ -1,11 +1,33 @@
-from flask import Flask
-
+from flask import Flask, render_template, request
 from todo_app.flask_config import Config
+import requests
```
Comment (Jonesey13) [inline todo_app/app.py:3]: You are using the requests python dependency yet it is not listed in your `pyproject.toml` file and so this dependency will not be installed on developers' machines when running `poetry install`.

To fix this you should run `poetry add requests` from a terminal.

File: todo_app/data/trello_items.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from flask import Flask, render_template, request
+
+import requests
+import os
+
+def get_trello_cards():
+    url = f'https://api.trello.com/1/boards/{os.getenv("TRELLO_BOARD_ID")}/lists'
```
Comment (Jonesey13) [inline todo_app/data/trello_items.py:7]: These new Trello environment variables should also be mentioned in the `.env.template` file. You do mention this in the `README.md` but the exact variable names are not stated.

File: todo_app/app.py [hunk 3]
```diff
@@ -1,11 +1,33 @@
...
+                item = Item.from_trello_card(card, trello_list)
+                items.append(item)
+        return render_template('index.html', get_items=get_items)
+   
+    @app.route('/create-todo', methods=['Post'])
+    def create_new_todo():
+        response = trello_items.create_todo()
+        return index()
+        
+    @app.route('/update_status', methods=['POST'])
```
Comment (Jonesey13) [inline todo_app/app.py]: Inconsistent casing used on HTTP method: `Post` vs `POST`

File: todo_app/data/trello_items.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    print(os.getenv("API_KEY"))
+
+    querystring = {
+        "key":os.getenv("API_KEY"),
```
Comment (Jonesey13) [inline todo_app/data/trello_items.py]: (Minor Style Issue) Probably want a space after the semicolon here (a little crammed atm)

File: todo_app/data/trello_items.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+def create_todo():
+    url = "https://api.trello.com/1/cards"
+    new_todo_title = request.form['todo-name']
+    querystring = {
+        "key":os.getenv("API_KEY"),
+        "token":os.getenv("API_TOKEN"),
+        "idList":os.getenv("TRELLO_TODO_LIST_ID"),
+        "name": new_todo_title
+    }
+    response = requests.request("POST", url, params=querystring)
```
Comment (Jonesey13) [inline todo_app/data/trello_items.py:30]: If you're assigning the response to a variable here you should probably do something with the variable (e.g. check the response was successful via `response.raise_for_status()`)

File: test_app.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+from todo_app import app
+
+@pytest.fixture
+
+def client():
+    file_path = find_dotenv('.env.test')
+    load_dotenv(file_path, override=True)
+    test_app = app.create_app()
+    with test_app.test_client() as client:
+
```
Comment (Jonesey13) [inline test_app.py]: Unnecessary newline

File: test_get_items.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    #Arrange
+    items = [
+        Item("1", "A Test Todo", "To Do"),
+        Item("2", "An in progress Todo", "Doing"),
+        Item("3", "A completed Todo", "Done")
+    ]
+
+    get_items = ViewModel(items)
+    #Act
+    test_todo_items = get_items.todo_items
```
Comment (Jonesey13) [inline test_get_items.py:18]: `#Act` needs a newline before it (to clearly mark out the Arrange/Act/Assert sections)

Comment (Jonesey13) [reply test_get_items.py:18]: Also the `#Assert` section isn't declared anywhere

File: test_app.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import pytest, requests, os
+from dotenv import load_dotenv, find_dotenv
+from todo_app import app
+
+@pytest.fixture
+
+def client():
+    file_path = find_dotenv('.env.test')
```
Comment (Jonesey13) [inline test_app.py:8]: There is no `.env.test` file, i.e. this test likely works with real credentials.

**Review by Jonesey13 — CHANGES_REQUESTED** [review]
OK looks mostly there. This would pass an M1 project ex review but there are issues that prevent it passing M2 and M3 project ex reviews:
M2: `requests` missing as a dependency, details on how to provide the environment variables are incomplete (needs to be listed in the .env.template file as well)
M3: Missing an `.env.test` file

More details/suggestions are provided in the review comments below...

**Comment by lacattano** [comment]
LV= Classification: Internal

Thanks for the quick feedback Alex I’ll update and submit it as m2 evidence.

**Comment by JackMeadDev** [comment]
Changes since the previous review here reviewed on https://github.com/lacattano/DevOps-Course-Starter/pull/2, but basically feel free to merge this when you're ready!

**Pull request merged**